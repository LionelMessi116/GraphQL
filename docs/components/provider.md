### About

You want to use the _Provider_ component at the root of your project. This will configure your _urql_ client for use throughout the application.

### Props

| Name   | Type     | Default   | Description                                  |
| ------ | -------- | --------- | -------------------------------------------- |
| client | `Client` | undefined | The _urql_ client for use throughout the app |

### Example

```jsx
import { Provider, createClient } from 'urql';

// ...
const client = createClient({
  url: 'http://localhost:3001/graphql',
});

// ...
<Provider client={client}>
  <Home />
</Provider>;
```
