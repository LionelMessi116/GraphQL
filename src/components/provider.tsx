import React, { ReactNode } from 'react';
import { Client } from '../types';
import { ContextProvider } from './context';

/** Props for the [Provider]{@link Provider} component. */
export interface ProviderProps {
  /** Child components to be rendered. */
  children: ReactNode;
  /** Urql client returned from createClient. */
  client: Client;
}

/** Component to configure the [urql client]{@link Client} for use throughout the application. */
export const Provider: React.SFC<ProviderProps> = function(props) {
  return (
    <ContextProvider value={props.client}>{props.children}</ContextProvider>
  );
};
