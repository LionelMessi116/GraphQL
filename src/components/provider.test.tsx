import React from 'react';
import { Provider } from './provider';
import { shallow } from 'enzyme';

const client = { stub: true };
const children = <h1>This is a child element</h1>;

it('passes snapshot', () => {
  const wrapper = shallow(
    // @ts-ignore
    <Provider client={client}>{children}</Provider>
  );

  expect(wrapper).toMatchSnapshot();
});

it('mounts ContextProvider', () => {
  const wrapper = shallow(
    // @ts-ignore
    <Provider client={client}>{children}</Provider>
  );

  expect(wrapper.find('ContextProvider').length).toBe(1);
});

it('renders children', () => {
  const wrapper = shallow(
    // @ts-ignore
    <Provider client={client}>{children}</Provider>
  );

  expect(wrapper.find('h1').matchesElement(children)).toBeTruthy();
});
