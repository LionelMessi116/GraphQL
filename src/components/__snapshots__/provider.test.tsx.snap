// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "stub": true,
    }
  }
>
  <h1>
    This is a child element
  </h1>
</ContextProvider>
`;
